### PR TITLE
[Merged by Bors] - chore(*): forward-port #18947

### PR DESCRIPTION
**Pull request by ericrbg**
* [`ring_theory.ideal.quotient`@`2f39bcbc98f8255490f8d4562762c9467694c809`..`949dc57e616a621462062668c9f39e4e17b64b69`](https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/ideal/quotient?range=2f39bcbc98f8255490f8d4562762c9467694c809..949dc57e616a621462062668c9f39e4e17b64b69)
* [`data.polynomial.basic`@`2651125b48fc5c170ab1111afd0817c903b1fc6c`..`949dc57e616a621462062668c9f39e4e17b64b69`](https://leanprover-community.github.io/mathlib-port-status/file/data/polynomial/basic?range=2651125b48fc5c170ab1111afd0817c903b1fc6c..949dc57e616a621462062668c9f39e4e17b64b69)
* [`algebra.monoid_algebra.basic`@`f69db8cecc668e2d5894d7e9bfc491da60db3b9f`..`949dc57e616a621462062668c9f39e4e17b64b69`](https://leanprover-community.github.io/mathlib-port-status/file/algebra/monoid_algebra/basic?range=f69db8cecc668e2d5894d7e9bfc491da60db3b9f..949dc57e616a621462062668c9f39e4e17b64b69)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Polynomial/Basic.lean [hunk 1]
```diff
@@ -319,15 +318,15 @@ instance module {S} [Semiring S] [Module S R] : Module S R[X] :=
...
-instance smulCommClass {S₁ S₂} [Monoid S₁] [Monoid S₂] [DistribMulAction S₁ R]
-    [DistribMulAction S₂ R] [SMulCommClass S₁ S₂ R] : SMulCommClass S₁ S₂ R[X] :=
+instance smulCommClass {S₁ S₂} [SMulZeroClass S₁ R] [SMulZeroClass S₂ R] [SMulCommClass S₁ S₂ R] :
+  SMulCommClass S₁ S₂ R[X] :=
   ⟨by
     rintro m n ⟨f⟩
     simp_rw [← ofFinsupp_smul, smul_comm m n f]⟩
 #align polynomial.smul_comm_class Polynomial.smulCommClass
 
-instance isScalarTower {S₁ S₂} [SMul S₁ S₂] [Monoid S₁] [Monoid S₂] [DistribMulAction S₁ R]
```
Comment (eric-wieser) [inline Mathlib/Data/Polynomial/Basic.lean:329]: I don't think the `{S₁ S₂}` should be removed here

File: Mathlib/RingTheory/Ideal/Quotient.lean [hunk 2]
```diff
@@ -215,17 +215,29 @@ theorem exists_inv {I : Ideal R} [hI : I.IsMaximal] :
...
-computable inverses in some applications.
+/-- The quotient by a maximal ideal is a group with zero. This is a `def` rather than `instance`,
+since users will have computable inverses in some applications.
+
 See note [reducible non-instances]. -/
 @[reducible]
-protected noncomputable def field (I : Ideal R) [hI : I.IsMaximal] : Field (R ⧸ I) :=
+protected noncomputable def groupWithZero (I : Ideal R) [hI : I.IsMaximal] :
+    GroupWithZero (R ⧸ I) :=
   { Quotient.commRing I,
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Quotient.lean:225]: mathport says ` (by infer_instance : MonoidWithZero (R ⧸ I))`. Can you leave a porting note if that didn't work?

File: Mathlib/RingTheory/Ideal/Quotient.lean [hunk 3]
```diff
@@ -215,17 +215,29 @@ theorem exists_inv {I : Ideal R} [hI : I.IsMaximal] :
...
-    Quotient.isDomain I with
+    Quotient.isDomain
+      I with
     inv := fun a => if ha : a = 0 then 0 else Classical.choose (exists_inv ha)
-    mul_inv_cancel := fun a (ha : a ≠ 0) =>
-      show a * dite _ _ _ = _ by rw [dif_neg ha]; exact Classical.choose_spec (exists_inv ha)
+    mul_inv_cancel := fun a (ha : a ≠ 0) ↦
+      show a * dite _ _ _ = _ by rw [dif_neg ha] ; exact Classical.choose_spec (exists_inv ha)
```
Comment (eric-wieser) [inline Mathlib/RingTheory/Ideal/Quotient.lean:230]: These changes don't seem necessary to me; can you revert them?

Comment (ericrbg) [reply Mathlib/RingTheory/Ideal/Quotient.lean:230]: Sure, but Itthought using the ascii thing was the correct mathlib4 style

Comment (eric-wieser) [reply Mathlib/RingTheory/Ideal/Quotient.lean:230]: I think we decided not to change it during (forward) porting as it just adds noise. I don't think spaces around `;` is preferred at all.

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**